catroid/src/main/java/org/catrobat/catroid/ui/dialogs/OrientationDialog.java

```diff
@@ -135,12 +136,19 @@ protected void handleOkButtonClick() {
 		dismiss();
 	}
 
-	private void loadStageProject() {
-		String templateZipFileName =
-				createLandscapeProject ? templateData.landscapeZipFileName : templateData.portraitZipFileName;
+	private void downloadTemplateFile() {
+		String zipFileUrl = baseUrlForTemplates;
+		zipFileUrl += createLandscapeProject ? templateData.getLandscape() : templateData.getPortrait();
+
+		DownloadTemplateTask downloadTemplateTask = new DownloadTemplateTask(getActivity(), templateData, zipFileUrl);
```

Comment:
When getActivity() is called, it is safe to do so. The fragment is attached to its activity.
However, due to the LoadProject-AsyncTask the fragment may be detached from its activity. This would lead to getActivity() returning null. That's why the activity is saved in onAttach.